packages/nimble-components/src/table/specs/table-columns-hld.md

```diff
@@ -0,0 +1,329 @@
+
+### `TableCellState` interface
+
+A table cell represents a single column for a single row. The data that a cell has access to will be a subset of the data for the entire row. An instance of a table cell will be generic to describe the subset of data it contains, where the `TCellData` type is a superset of the type represented by [`TableRecord`](https://github.com/ni/nimble/blob/3e4b8d3dd59431d1671e381aa66052db57bc475c/packages/nimble-components/src/table/types.ts#L24):
+
+```TS
+interface TableCellState<TCellData extends TableRecord, TColumnConfig> {
+  data: TCellData;
+  columnConfig: TColumnConfig;
+}
+```
+
+This interface could possibly be expanded in the future to communicate relevant table state to the cell template, such as whether or not the row is selected.
+
+### `TableColumn<>`
+
+This abstract class is what a column web component (i.e. a slotted column element) must extend.
+
+```TS
+abstract class TableColumn<TCellData extends TableRecord = TableRecord, TColumnConfig = {}> {
+    // This method returns the relevant, static configuration a column rquires its cellTemplate
+    // to have access to
+    getColumnConfig(): TColumnConfig {}
+
+    // The template to use to render the cell content for the column
+    abstract cellTemplate: ViewTemplate<TableCellState<TCellData, TColumnConfig>>;
+
+    // The style to apply to the cellTemplate
+    cellStyles?: ElementStyles;
+
+    // The keys that should be present in TCellData.
+    // This array is parallel with the keys returned from `getRecordFieldNames()`.
+    readonly cellStateDataFieldNames: readonly string[];
+
+    // The keys from the row data that correlate to the data that will be in TCellData.
+    // This array is parallel with the keys specified by `cellStateDataFieldNames`.
+    abstract getRecordFieldNames(): string[];
+
+    // Function that allows the table column to validate the type that gets created
+    // for the cell data. This should validate that the types in TCellData are correct
+    // for each key defined by `cellStateDataFieldNames`.
+    // This function should throw if validation fails.
+    abstract validateCellData(cellData: TCellData): void;
+}
+```
+
+_Note: The `TableColumn` class may be updated to support other features not covered in this HLD such as sorting and grouping._
+
+Given the above class, a series of column elements to handle basic use cases can be written within Nimble. For example, the `TableColumn` implementation we could create for rendering data as a read-only `NimbleTextField` could look like this:
+
+```TS
+type TableColumnTextCellData = StringField<'value'>;
+type TableColumnTextColumnConfig = { placeholder: string };
+
+// this interface is used to pass auxiliary configuration to access within the cellTemplate
+interface TableColumnTextCellState<TCellData extends TableRecord, TableColumnTextColumnConfig> extends TableCellState<TCellData, TableColumnTextColumnConfig> { }
```

Comment:
I don't think this interface is needed anymore, correct?

Reply:
Correct!